crates/flox/src/commands/environment.rs

```diff
@@ -129,6 +132,53 @@ impl EnvironmentCommands {
                     .await?
             },
 
+            EnvironmentCommands::WipeHistory {
+                // TODO use environment_args.system?
+                environment_args: _,
+                environment,
+            } => {
+                let environment_name = environment.as_ref().map(|e| e.to_str().unwrap());
+                let environment_ref: environment_ref::EnvironmentRef =
+                    resolve_environment_ref::<GitCommandProvider>(
+                        &flox,
+                        "wipe-history",
+                        environment_name,
+                    )
+                    .await?;
+
+                let environment = environment_ref
+                    .to_env::<GitCommandProvider, NixCommandLine>(&flox)
+                    .await
+                    .context("Environment not found")?;
+
+                match environment {
+                    CommonEnvironment::Named(env) => {
+                        if env.delete_symlinks().await? {
+                            // The flox nix instance is created with `--quiet --quiet`
+                            // because nix logs are passed to stderr unfiltered.
+                            // nix store gc logs are more useful,
+                            // thus we use 3 `--verbose` to have them appear.
+                            let nix = flox.nix::<NixCommandLine>(vec![
+                                "--verbose".to_string(),
+                                "--verbose".to_string(),
+                                "--verbose".to_string(),
+                            ]);
+                            let store_gc_command = StoreGc {
+                                ..StoreGc::default()
+                            };
+
+                            info!("Running garbage collection. This may take a while...");
+                            store_gc_command.run(&nix, &Default::default()).await?;
```

Comment:
I think GC might actually clean up too much for the developer persona, i.e. dev shells and dependendencies will need to be recreated after a gc run.

This approach works for us and most flox users but we might look closer at using `nix store delete` to delete paths more selectively.